app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -74,48 +74,29 @@ private void storeStuct(IASTDeclaration[] declarations, String name) throws Exce
             }
         }
         structRepo.put(name, struct);
-    }
 
-    public void mapPayload(IASTNode[] comments) {
+    }
 
+    private void associateIDStruct(IASTNode[] comments) {
+        Pattern pattern = Pattern.compile("@canPayloadStruct\\s+(\\S+)\\s*=\\s*(\\S+)");
         for (IASTNode comment : comments) {
-            char[] line = simplifyString(comment.getRawSignature());
-            if (line[0] == '|') {
-                String idName = findString(line, 1);
-                int secondIndex = idName.length() + 2;
-                String structName = findString(line, secondIndex);
-                if (!idStruct.containsKey(structName)) {
-                    PayloadMap rep = new PayloadMap(structName);
-                    idStruct.put(structName, rep);
+            Matcher matcher = pattern.matcher(comment.getRawSignature());
+            if (matcher.find()) {
+                if (!idStruct.containsKey(matcher.group(2))) {
+                    ArrayList<String> list = new ArrayList<>();
+                    idStruct.put(matcher.group(2), list);
                 }
-                idStruct.get(structName).canIDNames.add(idName);
+                idStruct.get(matcher.group(2)).add(matcher.group(1));
             }
         }
     }
 
-    public char[] simplifyString(String line) {
-        line = line.replace("//", "");
-        line = line.replace(" ", "");
-        return line.toCharArray();
-    }
-
-    public String findString(char[] line, int start){
-        StringBuilder name = new StringBuilder();
-        char character = line[start];
-        while (character != '|') {
-            name.append(character);
-            start += 1;
-            character = line[start];
-        }
-        return name.toString();
-    }
-
-    //Given a CAN ID name, retrieves its associated struct
-    public ArrayList getAssociatedStruct(String idName) {
+    //Given a CAN ID name, retrieves its associated struct contents
+    public ArrayList<StructContents> getAssociatedStruct(String idName) {
         for (String struct : idStruct.keySet()) {
-            PayloadMap curr = idStruct.get(struct);
-            if (curr.canIDNames.contains(idName)) {
-                return getStruct(curr.struct);
+            ArrayList<String> curr = idStruct.get(struct);
```

Comment:
If you find yourself iterating over a hashmap to find a particular value, you may be doing your data structures wrong